src/Illuminate/Database/Eloquent/Relations/BelongsToMany.php

```
@@ -197,6 +198,9 @@ public function get($columns = ['*'])
      */
     public function paginate($perPage = null, $columns = ['*'], $pageName = 'page')
     {
+        if ($perPage < 0) {
+            throw new RuntimeException("Invalid argument perpage."); 
```

You need to throw an InvalidArgumentException.